File: Node/TCP_Server/RequestHandler.js
class RequestHandler 
{    
    constructor()
    {   
        this.requests = [];
        this.requestType = {
            'status' : 0,
            'data' : 1,
            'config_get' : 2,
            'config_set' : 3,
            'actions' : 4,
        } 
    }   

    addRequest(name, requestType) 
    {   
        //TODO:?: Check if a request already exists for this address 
        this.requests.push(this.newRequest(name, requestType))    
    }

    async checkForRequest(data)
    {             
        
        let name = 'Client101';
                 
        let req = this.requests.find(request => (request.name === name))    //Save request
        
        //console.log(req);
        
        if(req == null)
        {
            req = this.newRequest(name, 'status')       
        }else{              
            this.requests = this.requests.filter(request => (request.id !== req.id))  //Remove request
        }     
        
        return JSON.stringify(req)
    }

    removeRequest(id) 
    {
        //TODO: We need to check if this works
        this.requests = this.requests.filter(request => (request.id !== id)) //find for first entry
    }

    newRequest(name, requestType)
    {
        let request =
        {
            id : this.id = Math.random().toString(32).substr(2, 8),
            name : name,
            requestType : requestType
        }
        return request;
    }

    clearAllRequests()
    {
        console.log('Clearing all requests...')        
        this.requests = [];
    }
}

module.exports = RequestHandler;

File: ESP32/TCP_Client/src/RequestHandler.cpp
#include <RequestHandler.h>


RequestHandler::requestEnum RequestHandler::getRequestType(const char* requestType)
{
    requestEnum rType = Status;
    String rTypeString = requestType;
    
    if(rTypeString == "data")
    {
        rType = Data;
    }else if(rTypeString == "action")
    {
        rType = Action;
    }else if(rTypeString == "config_get")
    {
        rType = Config_get;
    }
    else if(rTypeString == "config_set")
    {
        rType = Config_set;
    }

    return rType;
}

void RequestHandler::handleRequest()
{
    //const char* name = Request["name"];   
    const char* requestType = RequestHandler::request["requestType"];  
    
    switch (RequestHandler::getRequestType(requestType))
    {
    case Status:
        Serial.println("Send Status");
        break;
    case Data:
        Serial.println("Send data");
        break;
    case Action:
        Serial.println("Trigger Action");
        break;
    case Config_get:
        Serial.println("Send config");
        break;
    case Config_set:
        Serial.println("Set config");
        break;    
    default:
        Serial.println("Unknown requestType received");
        break;
    }
}